packages/contracts/contracts/LenderCommitmentForwarder.sol

```diff
@@ -75,113 +129,206 @@ contract LenderCommitmentForwarder is TellerV2MarketForwarder {
+   * @param _commitmentId The id of the commitment to delete.
+   
      */
-    function _deleteCommitment(
-        address _lender,
-        uint256 _marketId,
-        address _tokenAddress
-    ) internal {
-        if (
-            lenderMarketCommitments[_lender][_marketId][_tokenAddress]
-                .maxPrincipal > 0
-        ) {
-            delete lenderMarketCommitments[_lender][_marketId][_tokenAddress];
-            emit DeletedCommitment(_lender, _marketId, _tokenAddress);
-        }
+    function _deleteCommitment(uint256 _commitmentId) internal {
+        require(
+            lenderMarketCommitments[_commitmentId].maxPrincipal > 0,
+            "Commitment with zero max principal cannot be deleted."
+        );
+
+        delete lenderMarketCommitments[_commitmentId];
+        emit DeletedCommitment(_commitmentId);
     }
 
     /**
      * @notice Reduces the commitment amount for a lender to a market.
-     * @param _lender The address of the lender of the commitment.
-     * @param _marketId The Id of the market the commitment removal applies to.
-     * @param _tokenAddress The address of the asset for which the commitment is being removed.
+     * @param _commitmentId The id of the commitment to modify.
      * @param _tokenAmountDelta The amount of change in the maxPrincipal.
      */
     function _decrementCommitment(
-        address _lender,
-        uint256 _marketId,
-        address _tokenAddress,
+        uint256 _commitmentId,
         uint256 _tokenAmountDelta
     ) internal {
-        lenderMarketCommitments[_lender][_marketId][_tokenAddress]
+        lenderMarketCommitments[_commitmentId]
             .maxPrincipal -= _tokenAmountDelta;
     }
 
     /**
      * @notice Accept the commitment to submitBid and acceptBid using the funds
+
+     * @param _commitmentId The id of the commitment being accepted.
      * @param _marketId The Id of the market the commitment removal applies to.
-     * @param _lender The address of the lender of the commitment.
-     * @param _tokenAddress The address of the asset for which the commitment is being removed.
-     * @param _principal The amount of currency to borrow for the loan.
+     
+  
+     * @param _principalAmount The amount of currency to borrow for the loan.
+     * @param _collateralAmount The amount of collateral to use for the loan.
+     * @param _collateralTokenId The tokenId of collateral to use for the loan if ERC721 or ERC1155.
+
      * @param _loanDuration The loan duration for the TellerV2 loan.
      * @param _interestRate The interest rate for the TellerV2 loan.
      */
     function acceptCommitment(
+        uint256 _commitmentId,
         uint256 _marketId,
-        address _lender,
-        address _tokenAddress,
-        uint256 _principal,
+        uint256 _principalAmount,
+        uint256 _collateralAmount,
+        uint256 _collateralTokenId,
```

Comment:
I don't think we need to pass in `_marketId` and `_collateralTokenId` as they are already stored on the commitment and the `require` checks below are just redundant